smartconfig/_registry.py

```diff
@@ -0,0 +1,84 @@
+"""
+Internal register used to reference entries.
+
+Attributes:
+    global_configuration: A mapping of dotted paths to a mapping of attribute names and its value.
+    configuration_for_module: A mapping of dotted paths to the corresponding entry.
+"""
+
+from typing import Dict, Optional
+
+import smartconfig
+from smartconfig.typehints import EntryType, YAMLStructure, _EntryMapping
+from .exceptions import ConfigurationError, ConfigurationKeyError
+
+global_configuration: YAMLStructure = {}
+configuration_for_module: Dict[str, "smartconfig.ConfigEntry"] = {}
+
+mapping_cache: Dict[str, _EntryMapping] = {}
+
+
+def _lookup_mapping(path: str) -> _EntryMapping:
+    """
+    Lookup a mapping node.
+
+    Will try to use the mapping_cache when possible.
+
+    Args:
+         path: Dotted path of the mapping to lookup.
+
+    Returns:
+        The entry mapping at the requested path.
+
+    Raises:
+        ConfigurationKeyError: No override exists for this path.
+    """
+    if path in mapping_cache:
+        return mapping_cache[path]
+
+    current_node = global_configuration
+
+    for node_name in path.split('.'):
+        if node_name not in current_node:
+            raise ConfigurationKeyError(f"No override exists for the path {path!r}.")
```

Comment:
This will fail if any node on the path isn't a mapping.

Reply:
So we have to check that each node is a mapping when walking the tree?

Reply:
No, we just need to catch the error and transform it into something that is clearer.

Reply:
I ended up using an `isintance` check to match the style used in the rest of the code. I don't mind switching it back to a try-except though.